go-controller/pkg/node/gateway_udn_test.go

```diff
@@ -1774,6 +1786,169 @@ func TestConstructUDNVRFIPRulesPodNetworkAdvertised(t *testing.T) {
 					table:    1010,
 					dst:      *ovntest.MustParseIPNet("ae70::/60"),
 				},
+			},
+			v4mode: true,
+			v6mode: true,
+		},
+	}
+	config.Gateway.V6MasqueradeSubnet = "fd69::/112"
+	config.Gateway.V4MasqueradeSubnet = "169.254.0.0/16"
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			g := NewWithT(t)
+			node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: nodeName,
+				},
+			}
+			config.IPv4Mode = test.v4mode
+			config.IPv6Mode = test.v6mode
+			cidr := ""
+			if config.IPv4Mode {
+				cidr = "100.128.0.0/16/24"
+			}
+			if config.IPv4Mode && config.IPv6Mode {
+				cidr += ",ae70::/60/64"
+			} else if config.IPv6Mode {
+				cidr = "ae70::/60/64"
+			}
+			nad := ovntest.GenerateNAD("bluenet", "rednad", "greenamespace",
+				types.Layer3Topology, cidr, types.NetworkRolePrimary)
+			ovntest.AnnotateNADWithNetworkID("3", nad)
+			netInfo, err := util.ParseNADInfo(nad)
+			g.Expect(err).ToNot(HaveOccurred())
+			mutableNetInfo := util.NewMutableNetInfo(netInfo)
+			mutableNetInfo.SetPodNetworkAdvertisedVRFs(map[string][]string{node.Name: {"bluenet"}})
+			ofm := getDummyOpenflowManager()
+			// create dummy gateway interface(Need to run this test as root)
+			err = netlink.LinkAdd(&netlink.Dummy{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: "breth0",
+				},
+			})
+			g.Expect(err).NotTo(HaveOccurred())
+			udnGateway, err := NewUserDefinedNetworkGateway(mutableNetInfo, node, nil, nil, nil, nil, &gateway{openflowManager: ofm})
+			g.Expect(err).NotTo(HaveOccurred())
+			// delete dummy gateway interface after creating UDN gateway(Need to run this test as root)
+			err = netlink.LinkDel(&netlink.Dummy{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: "breth0",
+				},
+			})
+			g.Expect(err).NotTo(HaveOccurred())
+			udnGateway.vrfTableId = test.vrftableID
+			udnGateway.isNetworkAdvertised = true
+			udnGateway.isNetworkAdvertisedToDefaultVRF = true
+			rules, delRules, err := udnGateway.constructUDNVRFIPRules()
+			g.Expect(err).ToNot(HaveOccurred())
+			for i, rule := range rules {
+				g.Expect(rule.Priority).To(Equal(test.expectedRules[i].priority))
+				g.Expect(rule.Table).To(Equal(test.expectedRules[i].table))
+				g.Expect(rule.Family).To(Equal(test.expectedRules[i].family))
+				if rule.Dst != nil {
+					g.Expect(*rule.Dst).To(Equal(test.expectedRules[i].dst))
+				} else {
+					g.Expect(rule.Mark).To(Equal(test.expectedRules[i].mark))
+				}
+			}
+			for i, rule := range delRules {
+				g.Expect(rule.Priority).To(Equal(test.deleteRules[i].priority))
+				g.Expect(rule.Table).To(Equal(test.deleteRules[i].table))
+				g.Expect(rule.Family).To(Equal(test.deleteRules[i].family))
+				g.Expect(*rule.Dst).To(Equal(test.deleteRules[i].dst))
+			}
+		})
+	}
+}
+
+func TestConstructUDNVRFIPRulesPodNetworkAdvertisedToNoneDefaultNetwork(t *testing.T) {
```

Comment:
nit: `TestConstructUDNVRFIPRulesPodNetworkAdvertisedToNonDefaultVRF`?
its very confusing to see pod network advertised to non default network?